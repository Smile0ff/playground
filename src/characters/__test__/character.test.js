import React from 'react';
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';

import Character from '../components/character';

const characterData = {
  birth_year: '57BBY',
  created: '2014-12-10T16:16:29.192000Z',
  edited: '2014-12-20T21:17:50.325000Z',
  eye_color: 'blue-gray',
  films: [],
  gender: 'male',
  hair_color: 'auburn, white',
  height: '182',
  homeworld: '',
  mass: '77',
  name: 'Obi-Wan Kenobi',
  skin_color: 'fair',
  species: [],
  starships: [],
  url: '',
  vehicles: []
};

test('render character component', () => {
  const component = renderer.create(<Character character={ characterData } />);

  let tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});
