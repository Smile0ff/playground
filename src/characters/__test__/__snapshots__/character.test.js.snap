// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render character component 1`] = `
<article
  className="character"
>
  <h2>
    Obi-Wan Kenobi
  </h2>
  <p>
    Gender: 
    male
  </p>
  <p>
    Birthday: 
    57BBY
  </p>
</article>
`;
